Application auth method policies (#13418)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/applicationAuthMethodPolicy.md

---
title: "Azure AD application authentication methods API overview | Public preview"
description: "Application authentication methods is how apps get tokens in Azure AD."
localization_priority: Normal
author: "madansr7"
ms.prod: "identity-and-sign-in"
doc_type: "conceptualPageType"
---

# Azure AD application authentication methods API overview | Public preview

> 
> Namespace: microsoft.graph
> 
> Application authentication methods are ways by which applications can get tokens in Azure Active Directory (AD). Methods such as certificates or passwords secrets can be used to get a token as the application.
> Application authentication method policies provide a way for IT admins to establish app development best practices on these methods for the applications in their organizations. For e.g. an admin can configure a policy to block password secrets altogether or create a rule to ensure no password secret exceeds a lifetime threshold set by the organization. These policies can be configured to block all application or service principal objects by setting the enforcement targeting based on the object creation date.
> 
> These policies allows organizations to take advantage of the new security hardening features in app management and gradually adopt them in their organization. Enforcement of the restrictions based on application or service principal created datetime provides a way for an organization to review their app security posture, inventory apps and enforce controls per their resourcing schedules. The date based approach allows scaling back to cover existing applications once new applications start to follow the guidance.
> 
> There are two types of policy controls:
> 
> - Tenant default policies that apply to all applications or service principals.
> - Application/Service Principal specific policies which provide the ability to include or exclude certain applications from the tenant policy.
> 

Reviewer: ```suggestion

Application authentication methods such as certificates and password secrets allow apps to acquire tokens to access data in Azure Active Directory (Azure AD). The policies allow IT admins to enforce best practices for how apps in their organizations use these app authentication methods. For example, an admin might configure a policy to block the use or limit the lifetime of password secrets, and use the creation date of the object to enforce the policy.

These policies allow organizations to take advantage of the new app security hardening features. By enforcing restrictions that are based on the application or service principal created date, an organization can review their current app security posture, inventory apps, and enforce controls per their resourcing schedules and needs. This approach using the created date allows the organization to enforce the policy for new applications and also apply it to existing applications.

There are two types of policy controls:

- Tenant default policies that apply to all applications or service principals.
- Resource-specific (application or service principal) policies that allow inclusion or exclusion of individual applications from the tenant default policy.

```

Author: updated.

> ## Organization default app management policy

Reviewer: ```suggestion
## Tenant default app management policy
```

Author: updated.

> 
> A tenant default policy is a single object that always exists and disabled by default. Tenant default policy object has the ability to enforce distinct restrictions on application vs service principal objects.
> It contains two properties `applicationRestrictions` and `servicePrincipalRestrictions` that allow targeting applications owned by the tenant (application objects) or applications provisioned from another tenant (service principal objects) separately. This provides flexibility to either lock down apps originating within a tenant or help raise the quality bar for apps provisioned from outside the tenant boundary.
> 

Reviewer: ```suggestion

A tenant default policy is a single object that always exists and is disabled by default. It's defined by the [defaultAppManagementPolicy](defaultappmanagementpolicy.md) resource and enforces restrictions on application vs service principal objects.

It contains the following two properties:
+ **applicationRestrictions** allows targeting applications owned by the tenant (application objects).
+ **servicePrincipalRestrictions** allows targeting provisioned from another tenant (service principal objects.

These properties allow the organization to either lock down apps that originate within a tenant or raise the quality bar for apps that are provisioned from outside the tenant boundary.

```

> ## Application/Service Principal policy

Reviewer: ```suggestion
## Resource-specific (application and service principal) policy
```

> Applications or Service principals that need to be excluded from a tenant policy or need a fine grained restrictions could be assigned one of the custom policies defined in app management policy collection. These are a collection of policies that might have varying restrictions or different enforcement dates in them.
> Policies created as part of this collection can then be assigned to a specific application or service principal. In a scenario where org default policy and resource policy exist together, policy applied to a resource takes precedence and does not inherit from the tenant policy.
> Every application or service object can have only one policy assigned to it.
> 
> > [!Note]
> > Neither the tenant default policies nor the app/SP specific policies block token issuance for existing applications. So, an application that does not meet the policy requirements will continue to work until it tries to update the application and add a new secret.

Reviewer: ```suggestion
Resource-specific policies are defined in the [appManagementPolicy](appmanagementpolicy.md) resource, which contains a collection of policies with varying restrictions or different enforcement dates. An admin can then assign one of these policies to an application or service principal, excluding them from the tenant default policy. 

When both the tenant default policy and resource-specific policy exist, the resource-specific policy takes precedence and the assigned application or service principal doesn't inherit from the tenant policy. Only one policy can be assigned to an application or service principal.

> [!Note]
> Neither the tenant default policies nor the resource-specific policies block token issuance for existing applications. Therefore, an application that doesn't meet the policy requirements will continue to work until it tries to update to add a new secret.
```

Author: updated.

## What restrictions can be managed in Microsoft Graph?

> The application authentication methods policy APIs offer the following restrictions:

Reviewer: ```suggestion
The application authentication methods policy API offers the following restrictions:
```

Author: updated.

> The application authentication methods policy APIs offer the following restrictions:
> 
> | Restriction      | Description                                           | Examples                                                                                     |
> | :--------------- | :---------------------------------------------------- | :------------------------------------------------------------------------------------------- |
> | passwordAddition | Restrict password secrets on applications altogether. | Block new passwords on applications created on or after '01/01/2019'.                        |
> | passwordLifetime | Enforce a max lifetime range for a password secret.   | Restrict all new password secrets to 30 days max on applications created after '01/01/2019'. |

Reviewer: ```suggestion
The application authentication methods policy API offers the following restrictions:

| Restriction name | Description                                   | Examples                                                                                               |
|:-----------------|:----------------------------------------------|:-------------------------------------------------------------------------------------------------------|
| passwordAddition | Restrict password secrets on applications.    | Block new passwords on applications created on or after '01/01/2019'.                                  |
| passwordLifetime | Enforce a max lifetime for a password secret. | Restrict all new password secrets to a maximum of 30 days for applications created after '01/01/2019'. |
```

Author: updated.

> ### Single vs Multi-tenant apps
> 
> Single tenant apps should have policy reference applied to the application object.
> To restrict multi-tenant apps homed in customer tenant, a policy is applied to the application object where as multi-tenant apps provisioned from another tenant are managed by applying the policy to servicePrincipal object.

Reviewer: ```suggestion
### Single vs. Multi-tenant apps

For single tenant apps, apply the policy to the application object. To restrict multi-tenant apps homed in a customer tenant, apply the policy to the application object. To restrict multi-tenant apps provisioned from another tenant, apply the policy to the servicePrincipal object.
```

Author: I feel it is important to have them as separate bullets. It's a very important call out.

Reviewer: Noted. Changed the description.

> ### Summary of key difference between Default policy and Resource policies

Reviewer: ```suggestion
### Summary of key differences between the tenant default policy and resource-specific policies
```

Author: updated.

> | Default policy                                                     | Application/Service Principal policy                                                                      |
> | ------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------- |
> | Policy always exists.                                              | Policy objects can be created/updated to override default policy.                                         |
> | Restrictions are disabled by default for app/SP.                   | Allows customization for Single tenant, Multi tenant(backing app in home tenant or provisioned apps).     |
> | Allows only single restriction object definition.                  | Allows multiple policy objects to be defined, but only one can be applied to a resource.                  |
> | Allows differentiation between app objects and service principals. | Policy can be applied to either application or service principal object type.                             |
> | Applies all restrictions configured for every app/SP.              | Applies only the restrictions configured in the resource policy and does not inherit from default policy. |
> 
> ## Next steps
> 
> - Try the API in the [Graph Explorer](https://developer.microsoft.com/graph/graph-explorer).

Reviewer: ```suggestion
| Default policy                                                             | Resource-specific policy                                                                                                                            |
|----------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------|
| Policy always exists.                                                      | Policy objects are created or updated to override the default policy.                                                                               |
| Restrictions are disabled by default for all resources.                    | Allows customization for single-tenant and multi-tenant resources.                                                                                  |
| Allows only single restriction object definition for all resources.        | Allows multiple policy objects to be defined, but only one can be applied to a resource.                                                            |
| Allows distinction of restrictions for app objects vs. service principals. | Policy can be applied to either an application or service principal object.                                                                         |
| Applies all restrictions configured to all apps or service principals.     | Applies only the restrictions configured in the resource policy to the specified app or service principal, and doesn't inherit from default policy. |

## Next steps

- [defaultApManagementPolicy](defaultappmanagementpolicy.md) resource type.
- [appManagementPolicy](appmanagementpolicy.md) resource type.
```

Author: done.
